### PR TITLE
lib: nrf_cloud: add nrf_cloud_coap_bytes_send

### PR DESCRIPTION
**Pull request by maxd-nordic**
This patch adds a method to nRF Cloud CoAP to send raw bytes. (tested on dev stage)

### CODE REVIEW

**Review by plskeggs — APPROVED** [review]
Looks fine.  Just needs a changelog entry.

File: subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap.c
Comment (jayteemo) [inline subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap.c:224]: can you add `#define COAP_D2C_RAW_RSC "/msg/d2c/raw"` just to be consistent with the other resource paths?

**Review by jayteemo — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]
Approved assuming you implement Justin's change suggestion

**Comment by glarsennordic** [comment]
adding DNM so that it doesn't accidentally get merged due to approvals

**Review by plskeggs — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:525]:
```suggestion
    * Support for raw transfers to the :c:func:`nrf_cloud_coap_bytes_send` function.
```

**Review by divipillai — APPROVED** [review]

**Pull request merged**
